transports/quic/src/lib.rs

```diff
@@ -94,6 +98,14 @@ pub enum Error {
     /// The [`Connecting`] future timed out.
     #[error("Handshake with the remote timed out.")]
     HandshakeTimedOut,
+
+    /// Error when `Transport::dial_as_listener` is called without an active listener.
+    #[error("Tried to dial as listener without an active listener.")]
+    NoActiveListenerForDialAsListener,
+
+    /// Error when holepunching for a remote is already in progress
+    #[error("Already punching hole for ({0:?}, {1:?}).")]
```

Comment:
Do we need to debug print the address and peer id? They both implement `fmt::Display`.